### PR TITLE
[bluetooth] Add support for RadonEye (BLE) device

### PR DESCRIPTION
**Pull request by petero-dk**
Similar to airthings, provide a specific bluetooth device implementation to handle radon detection via the RadonEye

Only tested with original RadonEye RD200, compiled jar is available here: 
[org.openhab.binding.bluetooth.radoneye-3.3.0-SNAPSHOT.jar.zip](https://github.com/openhab/openhab-addons/files/7808141/org.openhab.binding.bluetooth.radoneye-3.3.0-SNAPSHOT.jar.zip)

CI built jar can be downloaded here: https://openhab.jfrog.io/ui/native/libs-pullrequest-local/org/openhab/addons/bundles/org.openhab.binding.bluetooth.radoneye

### CODE REVIEW

File: bundles/org.openhab.binding.bluetooth.radoneye/src/main/java/org/openhab/binding/bluetooth/radoneye/internal/RadoneyeHandler.java [hunk 1]
```diff
@@ -55,8 +55,8 @@ protected void updateChannels(int[] is) {
             data = RadoneyeDataParser.parseRd200Data(is);
             logger.debug("Parsed data: {}", data);
             Number radon = data.get(RadoneyeDataParser.RADON);
+            logger.debug("Parsed data radon number: {}", radon);
             if (radon != null) {
-                logger.debug("Parsed data radon: {}", String.format("[radon=%d Bq/m3]", radon));
```
Comment (petero-dk) [inline bundles/org.openhab.binding.bluetooth.radoneye/src/main/java/org/openhab/binding/bluetooth/radoneye/internal/RadoneyeHandler.java]: This line fails silently.

**Comment by petero-dk** [comment]
@wborn this is ready now, I am running this in my "production" and completed all checks.

**Comment by wborn** [comment]
Thanks for the PR @petro-pk!

We only reference authors by their real name and only for non-trivial contributions. So I think you can safely remove:

```
* @author the authors of the airthings bluetooth binding
```

I've also requested a review from a few other bluetooth binding contributors. :slightly_smiling_face:

**Comment by petero-dk** [comment]
I have removed the reference, but I would not say their contribution is non-trivial as this contribution is literally a repurpose of the Airthings Binding :-)

**Comment by petero-dk** [comment]
Any news on the reviews? Anything I can do?

File: bundles/org.openhab.binding.bluetooth.radoneye/src/main/resources/OH-INF/thing/radoneye.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+			<bridge-type-ref id="bluegiga"/>
+			<bridge-type-ref id="bluez"/>
+		</supported-bridge-type-refs>
+
+		<label>RadonEye RD200</label>
+		<description>Indoor radon monitor</description>
+
+		<channels>
+			<channel id="rssi" typeId="rssi"/>
+
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.bluetooth.radoneye/src/main/resources/OH-INF/thing/radoneye.xml:19]:
```suggestion
```

File: bundles/org.openhab.binding.bluetooth.radoneye/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+| address                         | String  |         | Yes      | Bluetooth address of the device (in format "XX:XX:XX:XX:XX:XX") |
+| refreshInterval                 | Integer | 300     | No       | How often a refresh shall occur in seconds                      |
+
+## Channels
+
+Following channels are supported for `RadonEye` thing:
+
+| Channel ID         | Item Type                | Description                                 |
+| ------------------ | ------------------------ | ------------------------------------------- |
+| radon              | Number:Dimensionless     | The measured radon level                    |
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.bluetooth.radoneye/README.md:32]: In the XML, you define the channel as `Number:Density`. Which definition is right?

**Review by kaikreuzer — CHANGES_REQUESTED** [review]
Thanks @petero-dk for your contribution and your patience!
Code looks good to me, I just have one last question - see below.

**Comment by joerg1985** [comment]
I tried to use the binary, is it the current build? The thing shows a com error in the ui and the logs are not verry helpful:

```
20:13:44.632 [INFO ] [hab.event.ThingStatusInfoChangedEvent] - Thing 'bl
uetooth:radoneye_rd200:85429d11aa' changed from OFFLINE (COMMUNICATION_E
RROR): Read/write data from device failed to OFFLINE (COMMUNICATION_ERRO
R): Connecting to device failed
20:13:44.678 [INFO ] [hab.event.ThingStatusInfoChangedEvent] - Thing 'bl
uetooth:radoneye_rd200:85429d11aa' changed from OFFLINE (COMMUNICATION_E
RROR): Connecting to device failed to OFFLINE (COMMUNICATION_ERROR): Rea
d/write data from device failed
20:13:45.098 [INFO ] [hab.event.ThingStatusInfoChangedEvent] - Thing 'bl
uetooth:radoneye_rd200:85429d11aa' changed from OFFLINE (COMMUNICATION_E
RROR): Read/write data from device failed to OFFLINE (COMMUNICATION_ERRO
R): Connecting to device failed
20:13:45.112 [INFO ] [hab.event.ThingStatusInfoChangedEvent] - Thing 'bl
uetooth:radoneye_rd200:85429d11aa' changed from OFFLINE (COMMUNICATION_E
RROR): Connecting to device failed to OFFLINE (COMMUNICATION_ERROR): Rea
d/write data from device failed
20:13:45.988 [INFO ] [hab.event.ThingStatusInfoChangedEvent] - Thing 'bl
uetooth:radoneye_rd200:85429d11aa' changed from OFFLINE (COMMUNICATION_E
RROR): Read/write data from device failed to OFFLINE (COMMUNICATION_ERRO
R): Connecting to device failed
20:13:46.054 [INFO ] [hab.event.ThingStatusInfoChangedEvent] - Thing 'bl
uetooth:radoneye_rd200:85429d11aa' changed from OFFLINE (COMMUNICATION_E
RROR): Connecting to device failed to OFFLINE (COMMUNICATION_ERROR): Rea
d/write data from device failed
^C~
```

I am able to manually connect to the RD200 device (firmware 2.0.2)
```
[CHG] Controller B8:27:EB:C5:0E:34 Pairable: yes
[bluetooth]# scan on
Discovery started
[bluetooth]# connect 24:4C:AB:00:B6:7E
Attempting to connect to 24:4C:AB:00:B6:7E
[CHG] Device 24:4C:AB:00:B6:7E Connected: yes
Connection successful
[CHG] Device 24:4C:AB:00:B6:7E ServicesResolved: yes
[FR:RE22207111667]# info
Device 24:4C:AB:00:B6:7E (public)
        Name: FR:RE22207111667
        Alias: FR:RE22207111667
        Paired: no
        Trusted: no
        Blocked: no
        Connected: yes
        LegacyPairing: no
        UUID: Unknown                   (00001523-0000-1000-8000-00805f9
b34fb)
        UUID: Generic Access Profile    (00001800-0000-1000-8000-00805f9
b34fb)
        UUID: Generic Attribute Profile (00001801-0000-1000-8000-00805f9
b34fb)
        RSSI: -31
```

File: bundles/org.openhab.binding.bluetooth.radoneye/README.md (hunk 3 above)
Comment (petero-dk) [reply bundles/org.openhab.binding.bluetooth.radoneye/README.md]: Sorry for the ridicules delay in fixing this. 

I have checked how the Airthings does it, and it is now aligned with that - the number has a dimension and it is density. Also all the time there was a specific SI calculation so it should be all fine.

**Comment by petero-dk** [comment]
Also, I don't know why the CI build compiles all modules? It fails on some other module

**Comment by lsiepel** [comment]
Some commits are missing DCO. hope you can fix that.

**Review by lsiepel — APPROVED** [review]
LGTM

**Comment by petero-dk** [comment]
> Hope you can fix the DCO, otherwise LGTM

I tried, but apparently I am not smart enough

**Review by kaikreuzer — APPROVED** [review]
Thanks for the contribution, @petero-dk and thanks for the review, @lsiepel!

The DCO is fine, it is part of the first commit and I simply squash the commits when merging.

**Pull request merged**